signature/verify.go

```diff
@@ -0,0 +1,63 @@
+package signature
+
+import (
+	"bytes"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/pkg/errors"
+	"math/big"
+	"strconv"
+)
+
+func VerifySignature(signer, signature []byte, message string) error {
+	address, err := verifyMessage(message, signature)
+	if err != nil {
+		return errors.Wrap(err, "verify message")
+	}
+
+	if len(address) != len(signer) {
+		return errors.Errorf("invalid signer length. expected %d. actual %d.", len(address), len(signer))
+	}
+
+	if !bytes.Equal(address, signer) {
+		return errors.Errorf("wrong signer (expected %x != actual %x)", address, signer)
+	}
+
+	return nil
+}
+
+func verifyMessage(message string, signedMessage []byte) ([]byte, error) {
```

Comment:
```suggestion
func recoverSignerAddress(message string, signedMessage []byte) ([]byte, error) {
```
let's rename to this because it is what this function does